safekeeper/src/timeline.rs

```diff
@@ -589,6 +592,80 @@ impl Timeline {
         shared_state.get_peers(conf.heartbeat_timeout)
     }
 
+    /// Start streaming tar archive with timeline:
+    /// 1) stream control file under lock;
+    /// 2) hold off WAL removal;
+    /// 3) collect BasebackupContext to understand which WAL segments should be
+    ///    streamed.
+    ///
+    /// Basebackup streams data up to flush_lsn. To make this safe, we must
+    /// check that term doesn't change during the procedure, or we risk sending
+    /// mix of WAL from different histories. Term is remembered in the
+    /// BasebackupContext and checked in finish_basebackup.
+    /// Note that in the last segment some WAL higher than flush_lsn set
+    /// here might be streamed; that's fine as long as terms doesn't change.
+    ///
+    /// In the future we might send only up to commit_lsn to get some valid
```

Comment:
This is probably not true as we might lose committed data if we rely only on commit_lsn, we discussed it a week ago.

Reply:
Reformulated the comment.